x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WaitForRolloverReadyStep.java

```diff
@@ -88,11 +88,10 @@ public void evaluateCondition(Metadata metadata, Index index, Listener listener,
         DataStream dataStream = indexAbstraction.getParentDataStream();
         if (dataStream != null) {
             boolean isFailureStoreWriteIndex = index.equals(dataStream.getFailureStoreWriteIndex());
-            if (isFailureStoreWriteIndex == false && index.equals(dataStream.getWriteIndex()) == false) {
+            if (isFailureStoreWriteIndex == false && dataStream.getWriteIndex().equals(index) == false) {
                 logger.warn(
-                    "index [{}] is not the {}write index for data stream [{}]. skipping rollover for policy [{}]",
+                    "index [{}] is not the write index for data stream [{}]. skipping rollover for policy [{}]",
                     index.getName(),
-                    isFailureStoreWriteIndex ? "failure store " : "",
```

Comment:
Why do we not want to mention `failure store` here?

Reply:
With the new logic this boolean is always false inside of this code block

Reply:
Should we perhaps move
```
            targetFailureStore = dataStream.isFailureStoreIndex(index.getName());
```
a couple of lines up and use that for the exception message?